elyra/pipeline/kfp/processor_kfp.py

```diff
@@ -692,6 +696,14 @@ def _sanitize_operation_name(name: str) -> str:
         """
         return re.sub('-+', '-', re.sub('[^-_0-9A-Za-z ]+', '-', name)).lstrip('-').rstrip('-')
 
+    @staticmethod
+    def _sanitize_param_name(name: str) -> str:
+        """
+        Sanitize a component parameter name.
+        (this is how KFP ensures component inputs/outputs are valid python kwarg names)
+        """
+        return re.sub('[^_0-9a-z]+', '_', name.lower()).lstrip('_').rstrip('_')
```

Comment:
Can you also take care of names that start with a numeric? As is, the regex still produces an invalid Python identifier. Example: `123-or-more`

Reply:
@ptitzler thanks for raising this, but the issue was not with the regex, it was me being stupid and using the sanitization from Kubeflow's V2 API.

I have updated the PR to use the EXACT same sanitization as Kubeflow 1.X now.